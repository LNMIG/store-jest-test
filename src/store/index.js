import { configureStore } from '@reduxjs/toolkit';
import initialState from '../initialState/index';
import reducer from '../reducers/index.js';

export const store = configureStore(reducer, initialState);