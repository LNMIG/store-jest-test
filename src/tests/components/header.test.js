import React from "react"
import {render, screen} from '@testing-library/react'
import ProviderMock from "../mocks/providerMock"
import Header from '../../components/header'


import { store } from '../../store/index'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';


describe(('Testing Header component'), () => {
    test('Check Header is rendering', () => {
        render(
            <Provider store={store}>
                <BrowserRouter>
                    <Header/>
                </BrowserRouter>
            </Provider>
        )
        expect(screen.getByText('T-shirt Store')).toBeInTheDocument()
        expect(screen.getByRole('heading', {name: 'T-shirt Store'})).toHaveAttribute('class', 'Header-title')
        expect(screen.getByRole('link', {name: 'T-shirt Store'})).toHaveAttribute('href', '/')
        expect(screen.getByRole('link', {name: ''})).toHaveAttribute('href', '/checkout')
    });

    test('Check Header is rendering with ProviderMock', () => {
      render(
        <ProviderMock>
            <Header/>
        </ProviderMock>
      )
    })
})