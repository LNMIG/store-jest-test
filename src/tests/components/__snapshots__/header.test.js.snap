// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Check Header snapshot 1`] = `[Function]`;
